websockify/websocketserver.py

```diff
@@ -0,0 +1,872 @@
+#!/usr/bin/env python
+
+'''
+Python WebSocket server base with support for "wss://" encryption.
+Copyright 2011 Joel Martin
+Copyright 2016 Pierre Ossman
+Licensed under LGPL version 3 (see docs/LICENSE.LGPL-3)
+
+You can make a cert/key with openssl using:
+openssl req -new -x509 -days 365 -nodes -out self.pem -keyout self.pem
+as taken from http://docs.python.org/dev/library/ssl.html#certificates
+
+'''
+
+import os, sys, time, errno, signal, socket, select, logging
+from base64 import b64encode, b64decode
+
+# Imports that vary by python version
+
+# python 3.0 differences
+if sys.hexversion > 0x3000000:
+    s2b = lambda s: s.encode('latin_1')
+else:
+    s2b = lambda s: s      # No-op
+try:    from http.server import SimpleHTTPRequestHandler
+except: from SimpleHTTPServer import SimpleHTTPRequestHandler
+
+# Degraded functionality if these imports are missing
+for mod, msg in [('ssl', 'TLS/SSL/wss is disabled'),
+                 ('multiprocessing', 'Multi-Processing is disabled'),
+                 ('resource', 'daemonizing is disabled')]:
+    try:
+        globals()[mod] = __import__(mod)
+    except ImportError:
+        globals()[mod] = None
+        print("WARNING: no '%s' module, %s" % (mod, msg))
+
+if multiprocessing and sys.platform == 'win32':
+    # make sockets pickle-able/inheritable
+    import multiprocessing.reduction
+
+from websockify.websocket import WebSocket, WebSocketWantReadError, WebSocketWantWriteError
+
+class Base64WebSocket(WebSocket):
```

Comment:
I think we can probably drop this -- we decided a while ago to drop base64 support in noVNC, so I'm not sure we need base64 support in Websockify any longer